lib/ecto/adapters/sql.ex

```diff
@@ -38,19 +38,15 @@ defmodule Ecto.Adapters.SQL do
       ## Types
 
       @doc false
-      def autogenerate(:id), do: nil
-      def autogenerate(:embed_id), do: Ecto.UUID.autogenerate()
-      def autogenerate(:binary_id), do: Ecto.UUID.autogenerate()
+      def autogenerate(:id),        do: nil
+      def autogenerate(:embed_id),  do: Ecto.UUID.generate()
```

Comment:
After the changes I made the autogenerate in custom types returns the custom value, and not the database representation, so I'd say it makes more sense.

The downside, as you said is with things that we can generate directly in the database version. Maybe we should have something to short-circuit the first step in database dumping of those types, so we only keep the adapter-specific logic?